test/Microsoft.AspNet.Server.KestrelTests/EngineTests.cs

```diff
@@ -91,10 +92,14 @@ public void EngineCanStartAndStop(ServiceContext testContext)
         [FrameworkSkipCondition(RuntimeFrameworks.Mono, SkipReason = "Test hangs after execution on Mono.")]
         public void ListenerCanCreateAndDispose(ServiceContext testContext)
         {
+            testContext.FrameFactory = (connectionContext, remoteEP, localEP) =>
+            {
+                return new Frame<HttpContext>(new DummyApplication(App), connectionContext, remoteEP, localEP);
+            };
```

Comment:
Maybe create `TestServiceContext.App` so this can turn into `testContext.App = App;` The setter can do exactly what this above does.